自定义异常类

```python
from django.db import DataError
from rest_framework.response import Response
from rest_framework.views import exception_handler

def my_exception_handler(exc, context):
    """
    自定义异常函数
    :exc 本次发生的异常对象
    :context 本次发生异常时的上下文环境信息，字典
    所谓执行上下文就是python解释器在执行代码时保存在内存中的变量，函数，类，对象，模块，等一系列的信息组成的环境信息
    """
    response = exception_handler(exc, context)
    if response is None:
        """当前发生的异常drf没有进行处理"""
        # 判断异常是否时我们预期内的
        if isinstance(exc, ZeroDivisionError):
            print("in")
            response = Response({"detail": "0不能作为除数"})
        if isinstance(exc, DataError):
            response = Response({"detail": "存储异常"})
    return response
```



settings中注册

根据自身函数路径灵活配置

```python
# 自定义异常[全局配置]
'EXCEPTION_HANDLER': "drfdemo.exceptions.my_exception_handler",
```



视图中直接触发期望异常即可

```python
class Demo6APIView(APIView):
    def get(self, request):
        1/0
        return Response({"msg": "ok"})
```

